src/Dapr.Extensions.Configuration/DaprSecretStoreConfigurationProvider.cs

```diff
@@ -37,14 +38,37 @@ internal class DaprSecretStoreConfigurationProvider : ConfigurationProvider
 
         private readonly DaprClient client;
 
+        private readonly CancellationTokenSource? cancellationTokenSource;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="DaprSecretStoreConfigurationProvider"/>.
+        /// </summary>
+        /// <param name="store">Dapr Secre Store name.</param>
+        /// <param name="normalizeKey">Indicates whether any key delimiters should be replaced with the delimiter ":".</param>
+        /// <param name="secretDescriptors">The secrets to retrieve.</param>
+        /// <param name="client">Dapr client used to retrieve Secrets</param>
+        public DaprSecretStoreConfigurationProvider(
+            string store,
+            bool normalizeKey,
+            IEnumerable<DaprSecretDescriptor> secretDescriptors,
+            DaprClient client) : this(store, normalizeKey, secretDescriptors, client, null)
+        {            
+        }
+
         /// <summary>
         /// Creates a new instance of <see cref="DaprSecretStoreConfigurationProvider"/>.
         /// </summary>
         /// <param name="store">Dapr Secre Store name.</param>
         /// <param name="normalizeKey">Indicates whether any key delimiters should be replaced with the delimiter ":".</param>
         /// <param name="secretDescriptors">The secrets to retrieve.</param>
         /// <param name="client">Dapr client used to retrieve Secrets</param>
-        public DaprSecretStoreConfigurationProvider(string store, bool normalizeKey, IEnumerable<DaprSecretDescriptor> secretDescriptors, DaprClient client)
+        /// <param name="cancellationTokenSource">Optional <see cref="CancellationTokenSource"/> used to configure the timeout waiting for Dapr.</param>
+        public DaprSecretStoreConfigurationProvider(
+            string store,
+            bool normalizeKey,
+            IEnumerable<DaprSecretDescriptor> secretDescriptors,
+            DaprClient client,
+            CancellationTokenSource? cancellationTokenSource)
```

Comment:
Can you explain why this is a source and not a token?

Reply:
It just felt like an easier way to specify when things were null and use a default. We've switched it around now to use a `TimeSpan` so that we can still control the timeout and also properly dispose of the source.